### PR TITLE
build chart with app-build-suite

### PR DESCRIPTION
**Pull request by ubergesundheit**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,11 +1,13 @@
 version: 2.1
 orbs:
-  architect: giantswarm/architect@0.14.0
+  architect: giantswarm/architect@1.1.2
```
Comment (ubergesundheit) [inline .circleci/config.yml:3]:
```suggestion
  architect: giantswarm/architect@2.2.0
```

File: helm/azure-ad-pod-identity-app/Chart.yaml [hunk 2]
```diff
@@ -2,9 +2,11 @@ apiVersion: v1
...
 name: azure-ad-pod-identity-app
-namespace: kube-system
 version: [[ .Version ]]
 home: https://github.com/giantswarm/azure-ad-pod-identity-app
 sources:
   - https://raw.githubusercontent.com/giantswarm/azure-ad-pod-identity-app/v[[ .Version ]]/README.md
 icon: https://s.giantswarm.io/app-icons/1/png/azure-ad-pod-identity-app-light.png
+restrictions:
+  fixedNamespace: kube-system
+  compatibleProviders: ["azure"]
```
Comment (oponder) [inline helm/azure-ad-pod-identity-app/Chart.yaml:12]: Wow! 👍

**Comment by tomahawk28** [comment]
```
2021-03-02 14:58:15,905 app_build_suite.build_steps.helm ERROR: kube-linter run failed with exit code 1
2021-03-02 14:58:15,905 app_build_suite.build_steps.helm ERROR: helm/azure-ad-pod-identity-app/templates/nmi-daemonset.yaml: (object: default/azure-ad-pod-identity-app-nmi apps/v1, Kind=DaemonSet) container "nmi" has DROP capabilities: [], but does not drop capability "NET_RAW" which is required (check: drop-net-raw-capability, remediation: NET_RAW grants an application within the container the ability to craft raw packets, use raw sockets, and it also allows an application to bind to any address. Please specify to drop this capability in the containers under containers security contexts.)
2021-03-02 14:58:15,905 app_build_suite.build_steps.helm ERROR: 
2021-03-02 14:58:15,906 app_build_suite.build_steps.helm ERROR: helm/azure-ad-pod-identity-app/templates/nmi-daemonset.yaml: (object: default/azure-ad-pod-identity-app-nmi apps/v1, Kind=DaemonSet) container "nmi" does not have a read-only root file system (check: no-read-only-root-fs, remediation: Set readOnlyRootFilesystem to true in your container's securityContext.)
2021-03-02 14:58:15,906 app_build_suite.build_steps.helm ERROR: 
2021-03-02 14:58:15,906 app_build_suite.build_steps.helm ERROR: helm/azure-ad-pod-identity-app/templates/nmi-daemonset.yaml: (object: default/azure-ad-pod-identity-app-nmi apps/v1, Kind=DaemonSet) container "nmi" is privileged (check: privileged-container, remediation: Don't run your container as privileged unless required.)
2021-03-02 14:58:15,906 app_build_suite.build_steps.helm ERROR: 
2021-03-02 14:58:15,906 app_build_suite.build_steps.helm ERROR: helm/azure-ad-pod-identity-app/templates/nmi-daemonset.yaml: (object: default/azure-ad-pod-identity-app-nmi apps/v1, Kind=DaemonSet) container "nmi" is not set to runAsNonRoot (check: run-as-non-root, remediation: Set runAsUser to a non-zero number, and runAsNonRoot to true, in your pod or container securityContext. See https://kubernetes.io/docs/tasks/configure-pod-container/security-context/ for more details.)
2021-03-02 14:58:15,906 app_build_suite.build_steps.helm ERROR: 
2021-03-02 14:58:15,906 app_build_suite.build_steps.helm ERROR: helm/azure-ad-pod-identity-app/templates/mic-deployment.yaml: (object: default/azure-ad-pod-identity-app-mic apps/v1, Kind=Deployment) object has 2 replicas but does not specify inter pod anti-affinity (check: no-anti-affinity, remediation: Specify anti-affinity in your pod spec to ensure that the orchestrator attempts to schedule replicas on different nodes. You can do this by using podAntiAffinity, specifying a labelSelector that matches pods of this deployment, and setting the topologyKey to kubernetes.io/hostname. See https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#inter-pod-affinity-and-anti-affinity for more details.)
2021-03-02 14:58:15,906 app_build_suite.build_steps.helm ERROR: 
2021-03-02 14:58:15,906 app_build_suite.build_steps.helm ERROR: helm/azure-ad-pod-identity-app/templates/mic-deployment.yaml: (object: default/azure-ad-pod-identity-app-mic apps/v1, Kind=Deployment) container "mic" does not have a read-only root file system (check: no-read-only-root-fs, remediation: Set readOnlyRootFilesystem to true in your container's securityContext.)
2021-03-02 14:58:15,906 app_build_suite.build_steps.helm ERROR: 
2021-03-02 14:58:15,906 app_build_suite.build_steps.helm ERROR: helm/azure-ad-pod-identity-app/templates/mic-deployment.yaml: (object: default/azure-ad-pod-identity-app-mic apps/v1, Kind=Deployment) container "mic" is not set to runAsNonRoot (check: run-as-non-root, remediation: Set runAsUser to a non-zero number, and runAsNonRoot to true, in your pod or container securityContext. See https://kubernetes.io/docs/tasks/configure-pod-container/security-context/ for more details.)
2021-03-02 14:58:15,906 app_build_suite.build_steps.helm ERROR: 
2021-03-02 14:58:15,906 app_build_suite.build_steps.helm ERROR: Error: found 7 lint errors
```

From what I heard, we could turn this off if we want it, @piontec, right?

**Comment by piontec** [comment]
> From what I heard, we could turn this off if we want it, @piontec, right?
Put a file like this one in chart's dir and name the tests you want to exclude:
https://github.com/giantswarm/app-build-suite/blob/master/examples/apps/hello-world-app/.kube-linter.yaml

**Review by piontec — APPROVED** [review]
Looks good! Do we need to set catalog URL in `main.yaml` or will architect-orb do it?
I only wonder if we should skip mem-limit-set check - this is usually useful, but still I don't know this app.

**Comment by ubergesundheit** [comment]
> Do we need to set catalog URL in main.yaml or will architect-orb do it?

architect-orb specifies the catalog url through flag, so the value in the yaml file will be overwritten

**Pull request merged**